### PR TITLE
Gui comm server fixes

### PR DESCRIPTION
**Pull request by ChristopheLarchier**

### CODE REVIEW

**Review by adrien-coye — DISMISSED** [review]
I checked the code, it looks good to me 🤝

File: src/server/comm/guicommserver_mac.mm [hunk 1]
```diff
@@ -153,7 +153,6 @@ - (void)processQuery:(NSData *_Nonnull)query callback:(void (^_Nonnull)(NSData *
         // Write JSON on the channel
         NSString *queryStr = [[NSString alloc] initWithData:newQuery encoding:NSUTF8StringEncoding];
         self.wrapper->inBuffer += std::string([queryStr UTF8String]);
-        self.wrapper->inBuffer += "\n";
```
Comment (luc-guyot-infomaniak) [inline src/server/comm/guicommserver_mac.mm:156]: Which kind of problems is this line fixing? Maybe some of them are are already impacting users.

Comment (herve-er) [reply src/server/comm/guicommserver_mac.mm:156]: [guicommserver_mac.mm](https://github.com/Infomaniak/desktop-kDrive/pull/1353/files/2d8238f7ec29d4bdb96297eb2fa2bd5bb6a10f0f#diff-bc5b047e86700d6a79cc0ed5bd84045dc5e9b200a1c750e4744d0610b139d00c) is a part of the commlayer between the server and the new client. It is not use by the legacy client.

File: src/server/comm/guicommserver.cpp [hunk 2]
```diff
@@ -42,6 +42,10 @@ bool GuiCommChannel::sendMessage(const CommString &message) {
 bool GuiCommChannel::canReadMessage() {
     fetchDataToBuffer();
     _validJsonInBuffer = containsValidJson(_readBuffer, _inBufferJsonEndIndex);
+    if (!_validJsonInBuffer)
```
Comment (herve-er) [inline src/server/comm/guicommserver.cpp]: `bool GuiCommChannel::containsValidJson` can return false in normal cases (e.g. only part of the JSON is received while the rest is still being transmitted or if `GuiCommChannel::canReadMessage()` is called while the buffer is empty)

Could this be moved to `bool GuiCommChannel::containsValidJson`?



```cpp
bool GuiCommChannel::containsValidJson(const CommString &message, size_t &endIndex) const {
    endIndex = 0;
    if (message.empty() || (message[0] != '{' && message[0] != '[')) {
        return false;
    }
```

to
```cpp
bool GuiCommChannel::containsValidJson(const CommString &message, size_t &endIndex) const {
    endIndex = 0;
    if (message.empty()) {
        return false;
    }

    if ((message[0] != '{' && message[0] != '[')) {
        LOGW_ERROR(Log::instance()->getLogger(),
                           L"The channel contains an invalid JSON: " << CommonUtility::commString2WStr(_readBuffer));
        return false;
    }
```

File: src/server/comm/guicommserver_mac.mm (hunk 1 above)
Comment (adrien-coye) [reply src/server/comm/guicommserver_mac.mm:156]: This is fixing an issue impacting the XPC layer. 

The `XPCGuiRemoteProtocol` is used by:
- The new GUI for macOS. (where the issue was discovered)
- With the `kDriveFinderSync` AppDelegate.m:173

It is not used by the `kDriveLiteSync` in my testing.

The original issue was the JSON validation (src/server/comm/guicommserver.cpp) was not passing after the new GUI client made a second XPC call, as the JSON would start with `\n`

File: src/server/comm/guicommserver.cpp (hunk 2 above)
Comment (adrien-coye) [reply src/server/comm/guicommserver.cpp]: We could also add a small unit test to check the corner cases (only part of the JSON is received while the rest is still being transmitted / empty buffer)

Comment (ChristopheLarchier) [reply src/server/comm/guicommserver.cpp]: [eb24833](https://github.com/Infomaniak/desktop-kDrive/pull/1353/commits/eb248335153c6bb5c188c40bae88eff4f5984521)

Tests:
[1aaf0e4](https://github.com/Infomaniak/desktop-kDrive/pull/1353/commits/1aaf0e4a9ba7b00527587f9b8ae02ba326f15667)

Comment (adrien-coye) [reply src/server/comm/guicommserver.cpp]: All good for me.

**Review by adrien-coye — DISMISSED** [review]
All good for me, I wait for @herve-er approval to merge this PR. 🙌

**Review by herve-er — DISMISSED** [review]

**Review by adrien-coye — APPROVED** [review]
CI is GO now, so I re approve this PR
